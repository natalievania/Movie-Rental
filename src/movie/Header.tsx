import React, { useState } from "react";
import * as Icons from "heroicons-react";
import { createSearchParams, useNavigate } from "react-router-dom";
import ModalCart from "./ModalCart";

const Header = () => {
  const navigate = useNavigate();
  const [search, setSearch] = useState("");
  const [isOpen, setOpen] = useState<boolean>(false);
  return (
    <>
      <div className="header flex bg-black-800 w-full px-[24px] py-[16px] items-center justify-between flex-3">
        <img
          onClick={() => navigate("/home")}
          src="https://www.freepnglogos.com/uploads/star-wars-logo-0.png"
          className="w-[7%] cursor-pointer"
        />
        <div className="grid grid-cols-4 gap-20">
          <p className="text-white">Title</p>
          <p className="text-white">Movies</p>
          <p className="text-white">Top IMDB</p>
          <p className="text-white">Other</p>
        </div>
        <div className="flex gap-4 items-center">
          <div className="relative w-[300px]">
            <div className="absolute inset-y-0 left-0 flex items-center pl-3 pointer-events-none">
              <Icons.Search />
            </div>
            <input
              type="text"
              onInput={(e: any) => setSearch(e.target.value)}
              onKeyDown={(e: any) =>
                e.key == "Enter" &&
                navigate({
                  pathname: "/home",
                  search:
                    search !== ""
                      ? `?${createSearchParams({
                          search: search,
                        })}`
                      : undefined,
                })
              }
              id="movie-search"
              className="bg-gray-50 border border-gray-300 text-gray-900 text-sm rounded-lg focus:ring-blue-500 focus:border-blue-500 block w-full pl-10 p-2.5  dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500"
              placeholder="Search movie name..."
            />
          </div>
          <Icons.ShoppingBag
            className="border border-white p-1 rounded-lg cursor-pointer"
            width={38}
            height={38}
            onClick={() => setOpen(true)}
          />
        </div>
      </div>
      {isOpen && <ModalCart handleModal={setOpen} />}
    </>
  );
};

export default Header;
