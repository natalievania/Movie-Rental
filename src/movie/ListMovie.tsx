import React, { useEffect, useState } from "react";
import * as Icons from "heroicons-react";
import { useNavigate } from "react-router-dom";
import Loading from "../components/Loading";

const ListMovie = () => {
  const history = useNavigate();
  const usp =
    window.location.search.replace("?search=", "") == ""
      ? ""
      : window.location.search;
  const [listFilm, setListFilm] = useState([]);
  const [isLoading, setIsLoading] = useState<boolean>(false);

  const fetchFilms = async () => {
    setIsLoading(true);
    let results = await fetch(`https://swapi.dev/api/films/${usp}`);
    const data = await results.json();
    setListFilm(data?.results);
    setIsLoading(false);
  };

  useEffect(() => {
    fetchFilms();
  }, [usp]);

  return (
    <div
      className="movie-list px-[24px] flex overflow-x-auto space-x-8"
      id="movie-list"
    >
      {isLoading && <Loading />}
      {!isLoading &&
        listFilm?.map((data: any, index) => {
          return (
            <div
              className="group flex flex-col items-center w-[245px] cursor-pointer flex-shrink-0"
              key={index}
              onClick={() =>
                history("/detail", { state: { detailUrl: data?.url } })
              }
            >
              <div className="h-[400px] relative">
                <img
                  src="https://www.originalfilmart.com/cdn/shop/products/Star_wars_last_jedi_2017_advance_intl_original_film_art_a.jpg?v=1551894963"
                  className="h-[400px] rounded-md group-hover:opacity-30"
                />
                <Icons.PlayOutline
                  width={70}
                  height={70}
                  className="absolute inset-0 mx-auto my-auto group-hover:visible invisible"
                />
                <p className="absolute top-0 right-0 p-1 bg-black rounded-sm">
                  HD
                </p>
                <p className="absolute top-0 left-0 p-1 bg-slate-800 rounded-sm">
                  Episode {data?.episode_id}
                </p>
              </div>
              <p className="title transform group-hover:scale-150 group-hover:underline transition duration-500 ease-in-out text-lg font-bold py-1 px-4 rounded-lg">
                {data?.title}
              </p>
              <div className="flex justify-between w-full items-center">
                <p className="p-1 text-sm">
                  {data?.release_date.substring(0, 4)}
                </p>
                <span className="border-solid rounded-lg border-2 border-white p-1 text-center text-sm">
                  IDR 123.000
                </span>
              </div>
              <p className="text-sm text-justify line-clamp-3 w-full mt-2">
                {data?.opening_crawl}
              </p>
            </div>
          );
        })}
    </div>
  );
};

export default ListMovie;
