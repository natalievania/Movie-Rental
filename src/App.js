import React, { useState } from "react";
import "./App.scss";
import { BrowserRouter, Routes, Route, Navigate } from "react-router-dom";
import Home from "./movie/Home";
import Detail from "./movie/Detail";
import Header from "./movie/Header";
export const CartContext = React.createContext(null);

function App() {
  const [cart, setCart] = useState([]);
  const [openModal, setOpenModal] = useState(false);
  const handleOpenModal = () => {
    if (openModal) {
      document.body.style.overflow = "unset";
    } else {
      if (typeof window != "undefined" && window.document) {
        document.body.style.overflow = "hidden";
      }
    }
    setOpenModal(!openModal);
  };
  return (
    <BrowserRouter>
      <CartContext.Provider value={{ cart: cart, setCart: setCart }}>
        <Header setOpenModal={handleOpenModal} />
        <Routes>
          <Route path="/" element={<Navigate to="/home" replace />} />
          <Route
            path="/home"
            element={
              <Home openModal={openModal} setOpenModal={handleOpenModal} />
            }
          />
          <Route path="/detail" element={<Detail />} />
        </Routes>
      </CartContext.Provider>
    </BrowserRouter>
  );
}

export default App;
