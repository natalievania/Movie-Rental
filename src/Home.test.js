import React from "react";
import { shallow, mount } from "enzyme";
import ListMovie from "./movie/ListMovie";
import App from "./App";
import Header from "./movie/Header";
import { BrowserRouter } from "react-router-dom";

describe("Test Home Page", function () {
  it("should match elements", function () {
    const wrapper = shallow(<App />);
    expect(wrapper.getElements()).toMatchSnapshot();
  });

  it("should search movie", function () {
    const wrapper = mount(
      <BrowserRouter>
        <Header />
      </BrowserRouter>
    );
    const form = wrapper.find(".search-movie");
    form.simulate("click");
    expect(form.length).toEqual(1);
    form.simulate("change", { target: { value: "Star" } });
    expect(form.find(".search-movie").get(0).value).toEqual("Star");
  });
});
