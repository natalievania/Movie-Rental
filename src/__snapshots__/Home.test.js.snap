// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Home Page should match elements 1`] = `
Array [
  <BrowserRouter>
    <Context.Provider
      value={
        Object {
          "cart": Array [],
          "setCart": [Function],
        }
      }
    >
      <Header />
      <Routes>
        <Route
          element={<Home />}
          path="/home"
        />
        <Route
          element={<Detail />}
          path="/detail"
        />
      </Routes>
    </Context.Provider>
  </BrowserRouter>,
]
`;

exports[`Test Home Page should render list movies 1`] = `
Array [
  <BrowserRouter>
    <Context.Provider
      value={
        Object {
          "cart": Array [],
          "setCart": [Function],
        }
      }
    >
      <Header />
      <Routes>
        <Route
          element={<Home />}
          path="/home"
        />
        <Route
          element={<Detail />}
          path="/detail"
        />
      </Routes>
    </Context.Provider>
  </BrowserRouter>,
]
`;
